phases/04-computer-vision/20-image-retrieval-metric/code/main.py

```
@@ -0,0 +1,93 @@
+        x = protos[labels] + 0.15 * torch.randn(bs, dim)
+        return x, labels
+
+    enc = Encoder(in_dim=dim, emb_dim=64)
+    opt = torch.optim.Adam(enc.parameters(), lr=3e-3)
+
+    for step in range(200):
+        x, y = sample(48)
+        emb = enc(x)
+        pos_idx, neg_idx = semi_hard_negatives(emb, y)
+        loss = triplet_loss(emb, emb[pos_idx], emb[neg_idx])
+        opt.zero_grad(); loss.backward(); opt.step()
```

Ruff (0.15.10)

[error] Multiple statements on one line (semicolon)

(E702)